include/SugarObjects/templates/basic/Basic.php

```diff
@@ -82,76 +88,100 @@ public function get_summary_text()
      * @throws RuntimeException
      * @throws InvalidArgumentException
      */
-    public function getEmailAddressConfirmOptInTick($emailField)
+    public function getConfirmOptInTickFromSugarEmailAddressField($emailField)
     {
-        $this->getEmailAddressValidateArguments($emailField);
+        $this->validateSugarEmailAddressField($emailField);
 
-        global $sugar_config, $app_strings;
+        return $this->displayOptInFromSugarEmailAddressField($emailField);
+    }
 
-        $tickHtml = '';
-        $confirmOptIn = $this->getEmailAddressConfirmOptIn($emailField);
-        if ($confirmOptIn && $sugar_config['email_enable_confirm_opt_in']) {
-            $tickTitle = $app_strings['LBL_CONFIRM_OPT_IN_TITLE'];
-            $tickHtml = '<span class="confirm-opt-in-tick" title="' . $tickTitle . '">&#10004;</span>';
+
+    /**
+     * @global array $sugar_config
+     * @global \LoggerManager $log
+     * @param string $emailField
+     * @return string
+     */
+    public function getOptInStatusFromSugarField($emailField)
+    {
+        $emailAddress = $this->fromSugarEmailAddressField($emailField);
+
+
+        if($emailAddress !== null && !in_array($this->module_name, self::$doNotDisplayOptInTickForModule, true)) {
+            if($emailAddress->invalid_email) {
+                return 'INVALID_EMAIL';
+            }
+
+            if ($emailAddress->opt_out == "1") {
+                return 'OPT_OUT';
+            }
+
+            if ($emailAddress->confirm_opt_in == "1") {
+                return "OPT_IN_PENDING_EMAIL_CONFIRMED";
+            } elseif(!empty($emailAddress->opt_in_email_created)) {
+                return "OPT_IN_PENDING_EMAIL_SENT";
+            } elseif(empty($emailAddress->opt_in_email_created)) {
+                return "OPT_IN_PENDING_EMAIL_NOT_SENT";
+            }
         }
 
-        return $tickHtml;
+        // Otherwise
+        return 'UNKNOWN_OPT_IN_STATUS';
     }
 
     /**
      *
      * @global array $sugar_config
      * @global \LoggerManager $log
      * @param string $emailField
-     * @return boolean
+     * @return \EmailAddress|\SugarBean
```

Comment:
Update php doc: It doesn't return `SugarBean` object anymore.

Reply:
Updated PHP Doc. There is problem with the retrieve method. The code inspection flags this up, because the PHP Doc in the SugarBean::retrieve() returns a SugarBean.

I suppose it's more truthful to say this returns an EmailAddress.